pkg/prometheus/operator.go

```diff
@@ -853,13 +853,77 @@ func (c *Operator) destroyPrometheus(key string) error {
 	return nil
 }
 
+func (c *Operator) loadBasicAuthSecrets(mons map[string]*v1alpha1.ServiceMonitor, s *v1.SecretList) (map[string]interface{}, error) {
+
+	secrets := map[string]interface{}{}
+
+	for _, mon := range mons {
+
+		for _, ep := range mon.Spec.Endpoints {
+
+			if ep.BasicAuth != nil {
+
+				var username string
+				var password string
+
+				for _, secret := range s.Items {
+
+					if secret.Name == ep.BasicAuth.Username.Key {
+
+						if u, ok := secret.Data[ep.BasicAuth.Username.Name]; ok {
+							username = string(u)
+						} else {
+							return nil, fmt.Errorf("Secret password of servicemonitor %s not found.")
+						}
+
+					}
+
+					if secret.Name == ep.BasicAuth.Password.Key {
+
+						if p, ok := secret.Data[ep.BasicAuth.Password.Name]; ok {
+							password = string(p)
+						} else {
+							return nil, fmt.Errorf("Secret username of servicemonitor %s not found.",
+								mon.Name)
+						}
+
+					}
+				}
+
+				secrets[fmt.Sprintf("%v/%v", mon, ep)] = map[string]interface{}{
```

Comment:
Since we already have a [format of identifying a single endpoints config section](https://github.com/gianrubio/prometheus-operator/blob/88a25963c2a61e2103970a35ae35574039c4f3e3/pkg/prometheus/promcfg.go#L77) I think it would be best to reuse that here.